test/test_gpu3.cpp

```diff
@@ -6035,6 +6036,9 @@ TEST_F(NVFuserTest, FusionBroadcastPersistentReduction_CUDA) {
 // Repro for
 // https://github.com/csarofeen/pytorch/issues/2094
 TEST_F(NVFuserTest, FusionRepro2094_CUDA) {
+  // disable cast optimization in pre segmenter, which causes numerical issue on
+  // tests
+  optimization::OptimizationGroupGuard<optimization::PreSegmenter> guard(false);
```

Comment:
What does the issue looks like?

Reply:
```
    auto t14 = at::mul(t8, t13);
    auto t15 = t14.to(at::ScalarType::Half);
    auto t16 = t15.to(at::ScalarType::Float);
```
Reference implementation has this ^^^ which truncates the data type and I believe that's giving us the numerical issue there.